src/redux/books/books.js

```diff
@@ -39,20 +27,77 @@ const booksReducer = (state = initialState, action) => {
 };
 
 // ACTIONS CREATOR
-export const addBook = (title, author) => ({
-  type: ADD_BOOK,
-  payload: {
-    id: uuidv4(),
-    title,
-    author,
+const toArray = (data) => {
+  const bookArray = Object.entries(data).map(([key, val]) => {
+    const { title, author, category } = val[0];
+    return {
+      id: key,
+      title,
+      author,
+      category,
+    };
+  });
+
+  return bookArray;
+};
+
+export const displayBooks = createAsyncThunk(
+  DISPLAY_BOOKS,
+  async (post, { dispatch }) => {
+    const response = await fetch(requestedURL);
+    const data = await response.json();
+    const books = toArray(data);
+    if (books) {
+      dispatch({
+        type: DISPLAY_BOOKS,
+        payload: books,
+      });
+    }
+  },
+);
+
+export const addBook = createAsyncThunk(
+  ADD_BOOK,
+  async (book, { dispatch }) => {
+    const response = await fetch(requestedURL, {
+      method: 'POST',
+      body: JSON.stringify({
+        item_id: book.id,
+        title: book.title,
+        author: book.author,
+        category: book.category,
+      }),
+      headers: { 'Content-type': 'application/json; charset=UTF-8' },
+    });
+    if (response.status === 201) {
+      dispatch({
+        type: ADD_BOOK,
+        payload: book,
```

Comment:
- [ ] as you can see in the screenshot I'm unable to get the books or create a book and also there are bunch of 404 errors in the console returned back from the API so please make sure to fix that in order your app is working properly without any errors 
<img width="1436" alt="Screen Shot 2022-11-11 at 17 16 29" src="https://user-images.githubusercontent.com/61790587/201384138-8cdb5eff-6efd-42a5-9a5b-de09d255ef55.png">